### ISSUE TITLE
"source --bulldin" ???

### ISSUE THREAD

**Issue by bar-g** [oilshell/oil#1711]
Really liking e.g. the the math.ysh features. I just seem to stumble over
the contradiction that lies in `source --buildin math.ysh` (as in  #1700)
for the user.

Such a line seems to bring up several things:
* Obviously math needs to bo sourced, even though dubbed buldin.
* YSH is said to have a math "buildin", but it does not work out of the box.
* Why a buldin needs to be sourced?

I get that technically it's something optional, but "build into the binary".

I guess there could be some word that can convey that optional-load-in property
without raising contradictions. Or maybe these internal libs could be loaded
on demand?

**Comment by bar-g** [comment]
Maybe libs in a specific subdirectory could be loaded even by default?

**Comment by bar-g** [comment]
similar to https://github.com/oilshell/oil/pull/1700#issuecomment-1681778246 (for something like `--opt` or `--optin`):
```
buildins/
  std/
     func_foo.py
     proc_foo.py 
     foo.ysh
  opt/
     math.ysh
  internal/
     install_latest.ysh
```

Could still allow `source --opt internal/install_latest.ysh` in addition to `osh --latest`.

**Comment by andychu** [comment]
Yeah I would like to not require source, but it's a bit of technical issue now.   I don't want it to incur startup time for `ysh -c 'echo hi'`

So I guess one thing to do is to build the parsed representation into the C++ binary, which is possible, but some work

I think we want to get  a solid first version of YSH, ditch the Python binary

and then polish these things

**Comment by bar-g** [comment]
Oh, nice, auto-loading will take away the contradiction "pain" from the default use.

I guess the `source` flag is to still there to stay, though, e.g. for the really optional things, i.e. libs like the "sh-compat" or "strict-compatible". So releasing the new flag right away with a wording that matches that, something like `--optin` which reflects the optional nature, could still make sense.
